tidy3d/plugins/adjoint/components/simulation.py

```diff
@@ -177,9 +178,11 @@ def _output_monitors_colocate_false(cls, val):
 
     @pd.validator("subpixel", always=True)
     def _subpixel_is_on(cls, val):
-        """Assert subpixel is on."""
-        if not val:
-            raise AdjointError("'JaxSimulation.subpixel' must be 'True' to use adjoint plugin.")
+        """Assert dielectric subpixel is on."""
+        if (isinstance(val, SubpixelSpec) and isinstance(val.dielectric, Staircasing)) or not val:
+            raise AdjointError(
+                "'JaxSimulation.subpixel.dielectric' must not be 'Staircasing' to use adjoint plugin."
```

Comment:
people might be confused if `subpixel=False`. Maybe we can just say that 
```
'JaxSimulation.subpixel' must be 'True' or a specific 'SubpixelSpec' to use adjoint plugin.
```
?

Reply:
actually maybe what I wrote is also not specific enough, `a specific 'SubpixelSpec'` -> `a 'SubpixelSpec' with no dielectric staircasing`?